### PR TITLE
Feature: Update all capital pool unit tests to use ethers instead of truffle/web3

### PR DESCRIPTION
**Pull request by kyledewy**
## Context

No ticket number for this one.

## Changes proposed in this pull request

I noticed the pool unit tests were still using truffle/web3 so I updated them before writing more tests in the upgrade-capital-pool PR

## Test plan
No tests were added

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [x] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: lib/constants.js [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
 
 const PoolUintParamType = {
-  minPoolEth: hex('MIN_ETH'),
+  minPoolEth: hex('MIN_ETH'.padEnd(8, '\0')),
 };
 
 const PoolAddressParamType = {
-  swapOperator: hex('SWP_OP'),
-  priceFeedOracle: hex('PRC_FEED'),
+  swapOperator: hex('SWP_OP'.padEnd(8, '\0')),
```
Comment (kyledewy) [inline lib/constants.js:15]: This was reverting without the 0 padding. I wasn't sure if the other ones were ok, so I added it there also.

File: contracts/modules/capital/Pool.sol [hunk 2]
```diff
@@ -389,7 +448,9 @@ contract Pool is IPool, MasterAwareV2, ReentrancyGuard {
     require(ok, "Pool: Transfer failed");
 
     // Transfer cover assets. Start from 1 (0 is ETH)
-    uint coverAssetsCount = coverAssets.length;
+    uint coverAssetsCount = coverAssets.length;  // substract 1 as ETH is skiped
```
Comment (danoctavian) [inline contracts/modules/capital/Pool.sol:451]: typo skiped -> skipped

Also you are starting with an index of 1 below so it seems to me you shouldn't substract for ETH to be skipped. and you actually don't so the comment is misleading

File: contracts/modules/capital/Pool.sol [hunk 3]
```diff
@@ -77,44 +78,45 @@ contract Pool is IPool, MasterAwareV2, ReentrancyGuard {
     _;
   }
 
+
   constructor (
```
Comment (danoctavian) [inline contracts/modules/capital/Pool.sol:82]: the constructor is quite complex now.

It deserves its own unit test at this point

**Review by danoctavian — CHANGES_REQUESTED** [review]
small change requests, mainly to add a dedicated constructor unit test

File: contracts/modules/capital/Pool.sol (hunk 2 above)
Comment (kyledewy) [reply contracts/modules/capital/Pool.sol:451]: Oh nice catch! Yup that comment didn't get deleted from the old code. Will fix that today.

File: contracts/modules/capital/Pool.sol (hunk 3 above)
Comment (kyledewy) [reply contracts/modules/capital/Pool.sol:82]: Yea I agree on this point. I wanted to update to ethers before writing the tests though, so maybe I will add them to the PR this is merged into?

File: test/unit/StakingPool/depositTo.js [hunk 4]
```diff
@@ -474,7 +474,7 @@ describe('depositTo', function () {
 
     const { firstActiveTrancheId, maxTranche } = await getTranches(daysToSeconds(0), DEFAULT_GRACE_PERIOD);
 
-    const tranches = Array(maxTranche - firstActiveTrancheId + 1)
+    const tranches = Array(maxTranche - firstActiveTrancheId)
```
Comment (kyledewy) [inline test/unit/StakingPool/depositTo.js:477]: Not sure why this broke during this PR, but the StakingPool was reverting on this test when using 8 tranches. The js variable `firstActiveTrancheId` isn't the current firstActiveTrancheId in the contract, but is the first one available for this deposit, relative to `period` and `gracePeriod`. It was iterating past the max active tranches for this reason.

File: contracts/modules/capital/Pool.sol (hunk 3 above)
Comment (danoctavian) [reply contracts/modules/capital/Pool.sol]: oh yeah sure where it make sense

File: test/utils/errors.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+// Used for generic compiler inserted panics.
+const COMPILER = 0x00;
```
Comment (danoctavian) [inline test/utils/errors.js:2]: assuming these were not defined anywhere in ethers

great work on having them here

**Review by danoctavian — APPROVED** [review]
LGTM good cleanup

some minor comment

File: contracts/modules/capital/Pool.sol (hunk 3 above)
Comment (danoctavian) [reply contracts/modules/capital/Pool.sol]: let's do this in the other PR right

**Pull request merged**